### PR TITLE
`Development`: Always use braces for code blocks

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context
In order to keep code readability at a high level, we should always use code blocks (curly braces) for if statements and loops. 
This PR adds a code style check to enforce that and fixes the issue in the current code base.

### Steps for Testing
code reivew

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

**Review by dearjasmina — APPROVED** [review]
It hurts my eyes to see the previous version, thank you for updating the code :D

File: src/main/resources/templates/java/test/staticCodeAnalysisConfig/checkstyle-configuration.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 <!DOCTYPE module PUBLIC
     "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
-    "https://checkstyle.org/dtds/configuration_1_3.dtd">
+    "https://checkstyle.sourceforge.io/dtds/configuration_1_3.dtd">
```
Comment (b-fein) [inline src/main/resources/templates/java/test/staticCodeAnalysisConfig/checkstyle-configuration.xml:4]:
```suggestion
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
```
The sourceforge website is only the legacy one. checkstyle.org is the new one.

The published default config.xml uses the checkstyle.org domain (https://github.com/checkstyle/checkstyle/blob/master/config/checkstyle-checks.xml), and the documentation only links to checkstyle.sourceforge.io for older release information (https://checkstyle.org/index.html#Previous_Version_Documentation).

The URL structure of both pages is the same, the fix should be doable by search-and-replace.

Comment (Strohgelaender) [reply src/main/resources/templates/java/test/staticCodeAnalysisConfig/checkstyle-configuration.xml:4]: Thanks for pointing this out. The old links did not work and Google always pointed to the sourgeforce one, so I used them here.

**Review by b-fein — APPROVED** [review]

**Review by simon-hng — APPROVED** [review]
This PR really does add braces to code blocks.

(Superior programming languages enforce this in the language itself  🦀)

**Pull request merged**